### PR TITLE
[ARM][AArch64] Enable FEAT_FHM for Arm Neoverse N2

### PR DESCRIPTION
**Pull request by jthackray**
Correct an issue with Arm Neoverse N2 after it was
changed to a v9a core in change f576cbe44eabb8a5ac0af817424a0d1e7c8fbf85:

  * FEAT_FHM should be enabled for this core.

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64.td
Comment (efriedma-quic) [inline llvm/lib/Target/AArch64/AArch64.td:687]: This looks like it affects every CPU that declares support for armv9; is this correct for all of them?

Comment (jthackray) [reply llvm/lib/Target/AArch64/AArch64.td:687]: I will update this patch to leave FEAT_MEC unchanged for now, but we will probably (in future) disable it for all CPUs (and add a flag to enable), since it's a v9.2a and higher optional feature.

**Comment by vhscampos** [comment]
Please double check if the arch features are listed in sorted alphabetical order. If so, please put FHM and fp16fml in their correct places.

**Comment by jthackray** [comment]
> Please double check if the arch features should listed in sorted alphabetical order. If so, please put FHM and fp16fml in their correct places.

Sure, now ordered when in a sorted list (the lists in TargetParserTest.cpp are not already sorted).

**Comment by vhscampos** [comment]
Looks good to me, but please wait for the others' approval. Thank you

**Review by rgwott — APPROVED** [review]
LGTM.

**Pull request merged**
